table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+      }
+      return shadowedByIgnore(tableName, true);
+    }
+    boolean hasBypassedPartition = mBypassedPartitions.containsKey(tableName);
+    if (hasBypassedPartition) {
+      // no need to check mTableBypassMode here, bc if a table has partition specifications,
+      // it must be in inclusion mode (or none)
+      return shadowedByIgnore(tableName, true);
+    }
+    // the table is not listed in any way
+    if (mTableBypassMode == Mode.EXCLUDE) {
+      return shadowedByIgnore(tableName, true);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Checks if all partitions of a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be fully bypassed, false otherwise
+   * @see UdbFilterSpec#isBypassedTable(String)
+   */
+  public boolean isFullyBypassedTable(String tableName) {
+    boolean isFullyBypassed = matchedByTemplates(mBypassedTables, tableName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(mTableBypassMode, isFullyBypassed));
+  }
+
+  /**
+   * Checks if a table is ignored.
+   *
+   * @param tableName the table name
+   * @return true if the table is ignored, false otherwise
+   */
+  public boolean isIgnoredTable(String tableName) {
+    boolean isIgnored = matchedByTemplates(mIgnoredTables, tableName);
+    return handleInclusionExclusion(mIgnoreMode, isIgnored);
+  }
+
+  /**
+   * Checks by a partition's name if it should be bypassed.
+   *
+   * @param tableName the table name
+   * @param partitionName the partition name
+   * @return true if the partition should be bypassed, false otherwise
+   */
+  public boolean isBypassedPartition(String tableName, String partitionName) {
+    if (isFullyBypassedTable(tableName)) {
+      return true;
+    }
+    if (!mBypassedPartitions.containsKey(tableName)) {
+      return false;
+    }
+    Mode partitionMode = mBypassedPartitions.get(tableName).getFirst();
+    boolean isBypassed =
+        matchedByTemplates(mBypassedPartitions.get(tableName).getSecond(), partitionName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(partitionMode, isBypassed));
+  }
+
+  private static boolean handleInclusionExclusion(Mode mode, boolean selected) {
+    switch (mode) {
+      case INCLUDE:
+        return selected;
+      case EXCLUDE:
+        return !selected;
+      case NONE:
+      default:
+        return false;
+    }
+  }
+
+  /**
+   * Checks if a table is shadowed by ignoring when it is also bypassed.
+   * @param tableName the table name
+   * @param mightBeShadowedIfTrue the return value of {@link #isBypassedTable(String)}, etc
+   * @return false if the table is shadowed by ignoring, mightBeShadowedIfTrue if it is not
+   */
+  private boolean shadowedByIgnore(String tableName, boolean mightBeShadowedIfTrue) {
+    if (mightBeShadowedIfTrue && isIgnoredTable(tableName)) {
+      // todo(bowen): this warning is never triggered bc if a table is ignored,
```

Comment:
what needs to be changed here?

Reply:
The check for ignored tables happens before the `mount` call. The checks for bypassed tables are done in the `mount` method, and carried out against the non-ignored tables. Therefore, we would never know if a table is misconfigured to be ignored while it is actually meant to be bypassed, since it is filtered out for the check.

I'm thinking of doing the check in the `isIgnoredTable` method, instead of doing it in the bypass methods.

Reply:
Done in 81a1fd41a67bf67c1dd3839cb5103f74f172f1a6